### PR TITLE
Add support for mixed constraints with constrained planner

### PR DESCRIPTION
**Pull request by stephanie-eng**
### Description

Adds support for using mixed constraint types (position and orientation) with OMPL's constrained planner.

Tutorial here:  https://github.com/ros-planning/moveit2_tutorials/pull/421

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_planners/ompl/ompl_interface/src/detail/ompl_constraints.cpp [hunk 1]
```diff
@@ -376,7 +376,7 @@ std::shared_ptr<BaseConstraint> createOMPLConstraint(const moveit::core::RobotMo
 
   // This factory method contains template code to support different constraints, but only position constraints are
   // currently supported. The other options return a nullptr for now and should not be used.
```
Comment (AndyZe) [inline moveit_planners/ompl/ompl_interface/src/detail/ompl_constraints.cpp:378]:
```suggestion
  // This factory method contains template code to support position and/or orientation constraints.
  // If the specified constraints are invalid, a nullptr is returned.
```

**Review by AndyZe — COMMENTED** [review]
I haven't tested it yet. The code changes look good and surprisingly simple though  :+1:

**Review by AdamPettinger — COMMENTED** [review]
Seems very straightforward and elegant. I played with this a bunch starting from https://github.com/ros-planning/moveit2_tutorials/pull/421 as an example, and it looks really good! Namely: single orientation, single position (with volume == 0 and volume > 0), and 1 position + 1 orientation all seem to be planning successfully very quickly.

**Review by AndyZe — APPROVED** [review]

**Comment by jih189** [comment]
The code looks strange to me. In the planning_context_manager.cpp, line 539 to line [544,](https://github.com/stephanie-eng/moveit2/blob/68da67770605628c1043abaf9c60c2d331951036/moveit_planners/ompl/ompl_interface/src/planning_context_manager.cpp#L542), it still said "Mixed constraints are not supported", and it only allows one constraint based on the if statement.
`  if (constrained_planning_iterator != pc->second.config.end() &&
      boost::lexical_cast<bool>(constrained_planning_iterator->second) &&
      ((req.path_constraints.position_constraints.size() == 1) !=
       (req.path_constraints.orientation_constraints.size() == 1)))`

I think it is wrong. If so, what have you tested?

**Pull request merged**